File: String/TypeScript/_1768_MergeStringsAlternately.ts
function mergeAlternately(word1: string, word2: string): string {
  let maxLength = word1.length > word2.length ? word1.length : word2.length;
  let res = "";
  for (let i = 0; i < maxLength; i++) {
    if (word1[i]) res += word1[i];
    if (word2[i]) res += word2[i];
  }
  return res;
}

console.log("Expected: apbqcr", "Output:", mergeAlternately("abc", "pqr"));
console.log("Expected: apbqrs", "Output:", mergeAlternately("ab", "pqrs"));
console.log("Expected: apbqcd", "Output:", mergeAlternately("abcd", "pq"));

File: String/Python/_1768_MergeStringsAlternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l1, l2 = len(word1), len(word2)
        max = l1 if l1 > l2 else l2
        res = ""
        for i in range(max):
            if l1 > i:
                res += word1[i]
            if l2 > i:
                res += word2[i]

        return res


ins = Solution()

print("Expected: apbqcr", "Output:", ins.mergeAlternately("abc", "pqr"))
print("Expected: apbqrs", "Output:", ins.mergeAlternately("ab", "pqrs"))
print("Expected: apbqcd", "Output:", ins.mergeAlternately("abcd", "pq"))
